device/sycl/src/clusterization/component_connection.sycl

```diff
@@ -5,86 +5,54 @@
  * Mozilla Public License Version 2.0
  */
 
-// SYCL library include(s)
+// SYCL library include(s).
 #include "../utils/get_queue.hpp"
 
 // SYCL include(s).
 #include <CL/sycl.hpp>
 
 // Project include(s).
-#include "traccc/sycl/clusterization/component_connection.hpp"
-
-// Vecmem include(s).
-#include <vecmem/utils/sycl/copy.hpp>
+#include "component_connection.hpp"
 
 namespace traccc::sycl {
 
-component_connection::component_connection(vecmem::memory_resource& mr,
-                                           queue_wrapper queue)
-    : m_mr(mr), m_queue(queue) {}
-
-host_cluster_container component_connection::operator()(
-    const host_cell_container& cells_per_event) const {
-
-    // Create the output container
-    output_type clusters(&m_mr.get());
+void component_connection(
+    cluster_container_view clusters_view, cell_container_const_view cells_view,
+    const host_cell_container& cells_per_event,
+    vecmem::data::vector_view<unsigned int> clusters_count_view,
+    vecmem::data::vector_view<unsigned int> total_clusters_view,
+    vecmem::memory_resource& resource, queue_wrapper queue) {
 
     // Execution size of the algorithm
     unsigned int N = cells_per_event.size();
```

Comment:
Please use `std::size_t` for size variables instead of `unsigned int` or `unsigned long`.

Generally `unsigned int` should only be used when we want to handle that variable atomically on the device.

Reply:
done